packages/contracts-bedrock/scripts/merkle-trie-fuzzer/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethdb/memorydb"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+// Represents a test case for bedrock's `MerkleTrie.sol`
+type TrieTestCase struct {
+	Root  *[32]byte
+	Key   []byte
+	Value []byte
+	Proof [][]byte
+}
+
+// Tuple type to encode `TrieTestCase`
+var (
+	trieTestCase, _ = abi.NewType("tuple", "TrieTestCase", []abi.ArgumentMarshaling{
+		{Name: "root", Type: "bytes32"},
+		{Name: "key", Type: "bytes"},
+		{Name: "value", Type: "bytes"},
+		{Name: "proof", Type: "bytes[]"},
+	})
+
+	encoder = abi.Arguments{
+		{Type: trieTestCase},
+	}
+)
+
+// Encodes the TrieTestCase as the `trieTestCase` tuple.
+func (t *TrieTestCase) AbiEncode() string {
+	// Encode the contents of the struct as a tuple
+	packed, err := encoder.Pack(&t)
+	if err != nil {
+		log.Fatalf("Error packing TrieTestCase: %v", err)
+	}
+
+	// Remove the pointer and encode the packed bytes as a hex string
+	return hexutil.Encode(packed[32:])
+}
+
+func main() {
+	// Create an empty merkle trie
+	memdb := memorydb.New()
+	randTrie := trie.NewEmpty(trie.NewDatabase(memdb))
+
+	// Seed the random number generator with the current unix timestamp
+	rand.Seed(time.Now().UnixNano())
```

Comment:
If this runs in parallel, is this enough entropy for it to be a good seed? Some runs might end up with the same time. I know that foundry fuzzing is good but not the greatest in the world when it comes to generating input values, perhaps taking the input value produced by foundry and hashing it with the time could produce better randomness?

Reply:
Good spot- changed from `math/rand` to `crypto/rand` rather than passing hashed values as an argument to the binary.